src/Mapper/UserHydrator.php

```diff
@@ -61,7 +61,9 @@ public function hydrate(array $data, $object)
             throw new Exception\InvalidArgumentException('$object must be an instance of LmcUser\Entity\UserInterface');
         }
 
+      if (isset($data['user_id'])) {
         $data = $this->mapField('user_id', 'id', $data);
+      }
```

Comment:
Although this change fixes the warning, this is not the right place to make the change.  If, for some reason, we needed to use the mapField() function with another key that does not exist in the data, we will get the warning again.
The change should be made in the mapField() function.
````
if (isset($array[$keyFrom])) {
  $array[$keyTo] = $array[$keyFrom];
  unset($array[$keyFrom];
}

return $array;
````